pruner/pruner.go

```diff
@@ -0,0 +1,339 @@
+package pruner
+
+import (
+	"sync"
+	"time"
+
+	quamina "quamina/lib"
+)
+
+// Stats reports basic counts to aid in deciding when to Rebuild.
+type Stats struct {
```

Comment:
A bit troubling that the word "pattern" doesn't appear in the name either of the type nor the fields.  When I'm reading code it seems like it would be nice to see things like `if stats.patternsEmitted` or `if patternStats.emitted` or some such.  I have been accused of being overly fond of long names…

Reply:
I have been accused of being overly fond of exceeding short, cryptic names (even symbols like δ).

In this case, the package itself sort of conveys (to me) what `Stats` and `Emitted` refer to.
`PatternStats` isn't quite right IMO since the data includes stuff not related to patterns (like `RebuildDuration`). The package is `pruner`, so `PrunerStats` seems redundant. 

`Live`, `Added`, `Emitted`, and `Filtered` all refer to patterns, which is a `Pruner`'s stock-in-trade, so adding the suffix `Pattern` to each also seems redundant.

All that said, I can suppress my character flaws for the greater good.